app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -154,15 +154,23 @@ protected void onSaveInstanceState(Bundle savedInstanceState) {
      *                       upon successful login
      */
     protected void initiateLoginAttempt(boolean restoreSession) {
-        if (uiController.getEnteredPasswordOrPin().equals("")) {
-            raiseLoginMessage(StockMessages.Auth_EmptyPassword, false);
-            return;
+        LoginMode loginMode = uiController.getLoginMode();
+
+        if (uiController.getEnteredPasswordOrPin().equals("") &&

```

Comment:
I generally like to the `equals` call to be made on the hard-coded string (`""`) if possible to avoid any potential for a null pointer exception.